src/libp2p_crypto.erl

```diff
@@ -299,6 +308,34 @@ to_network(?NETTYPE_TEST) -> testnet.
 %% @doc Verifies a binary against a given digital signature over the
 %% sha256 of the binary.
 -spec verify(binary(), binary(), pubkey()) -> boolean().
+verify(Bin, Signature, {multisig_m_of_n, M, N, KeysDigest}) ->
+    KeySize = 32, % TODO Macro?
```

Comment:
I think it'd be wise to have a recursive pattern match function here matching off the KeyType/KeyBytes N times and doing bin_to_pubkey as you go

Reply:
```erlang
parse_keys(Rest, 0, Acc) ->
    {lists:reverse(Acc), Rest};
parse_keys(<<NetType:4, ?KEYTYPE_ECC_COMPACT:4, PubKey:32/binary, Rest/binary>>, KeysRemaining, Acc) when KeyType ==?KEYTYPE_ECC_COMPACT; KeyType == ?KEYTYPE_ED25519 ->
    parse_keys(Rest, KeysRemaining -1, [bin_to_pubkey(<<<<NetType:4, KeyType:4, PubKey:32/binary)|Acc]).